### PR TITLE
[configuration] add configuration to skip blacklist check

### PR DESCRIPTION
**Pull request by luxe**
In response to #801
Example configs keep the current behavior via `use_deny_list: True`

### CODE REVIEW

File: src/main/java/build/buildfarm/instance/shard/ShardInstance.java [hunk 1]
```diff
@@ -2563,4 +2567,11 @@ public void deregisterWorker(String workerName) {
       throw Status.fromThrowable(e).asRuntimeException();
     }
   }
+
+  private boolean isInDenyList(RequestMetadata requestMetadata) throws IOException {
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/ShardInstance.java:2571]: Prefer 'inDenyList' here

File: examples/shard-server.config.example [hunk 2]
```diff
@@ -49,6 +49,10 @@ instances {
     # a fallback to non-remote recovery, rather than a catastrophic
     # failure.
     max_blob_size: 4294967296
+    
+    # Whether the instance should consult a denylist when looking up actions & invocations.
+    # For cache-only systems, users may want to disable this check.
```
Comment (werkt) [inline examples/shard-server.config.example:54]: Drop the comma

File: src/main/java/build/buildfarm/instance/shard/ShardInstance.java [hunk 3]
```diff
@@ -298,6 +298,7 @@
   private boolean stopping = false;
   private boolean stopped = true;
   private Thread prometheusMetricsThread;
+  private boolean useDenyList = true;
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/ShardInstance.java:301]: This can be final, no? Move it up if so.

**Review by werkt — APPROVED** [review]

File: src/main/java/build/buildfarm/instance/shard/ShardInstance.java (hunk 1 above)
Comment (luxe) [reply src/main/java/build/buildfarm/instance/shard/ShardInstance.java]: fixed

File: examples/shard-server.config.example (hunk 2 above)
Comment (luxe) [reply examples/shard-server.config.example]: fixed

File: src/main/java/build/buildfarm/instance/shard/ShardInstance.java (hunk 3 above)
Comment (luxe) [reply src/main/java/build/buildfarm/instance/shard/ShardInstance.java]: fixed

**Pull request merged**